app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/BgGraphBuilder.java

```diff
@@ -656,7 +680,7 @@ public LineChartData previewLineData(LineChartData hint) {
                 lines.add(subLine); // iob line
             }
 
-            predictive_end_time = simple ? end_time : ((end_time * FUZZER) + (60000 * 10) + (1000 * 60 * 60 * predictivehours)) / FUZZER; // used first in ideal/highline
+            predictive_end_time = simple ? end_time : timeStampToGraphPos(graphPosToTimestamp(end_time) + (60000 * 10) + (1000 * 60 * 60 * predictivehours)); // used first in ideal/highline
```

Comment:
To get a more readable code, we could use constants for the different millisecond spans.